ext/standard/array.c

```diff
@@ -5905,62 +5905,95 @@ PHP_FUNCTION(array_rand)
 /* {{{ Returns the sum of the array entries */
 PHP_FUNCTION(array_sum)
 {
-	zval *input,
-		 *entry,
-		 entry_n;
+	HashTable *input;
+	zval *entry;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_ARRAY(input)
+		Z_PARAM_ARRAY_HT(input)
 	ZEND_PARSE_PARAMETERS_END();
 
-	ZVAL_LONG(return_value, 0);
+	if (zend_hash_num_elements(input) == 0) {
+		// TODO Deprecate empty array
+		RETURN_LONG(0);
+	}
 
-	ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(input), entry) {
-		if (Z_TYPE_P(entry) == IS_ARRAY || Z_TYPE_P(entry) == IS_OBJECT) {
-			continue;
+	ZVAL_LONG(return_value, 0);
+	ZEND_HASH_FOREACH_VAL(input, entry) {
+		zend_result status = add_function(return_value, return_value, entry);
```

Comment:
why don't you use the `fast_add_function` anymore? It seems to be handling params well, while handling the most possible cases faster.

Reply:
Because ``add_function`` is defined as:
```c

ZEND_API zend_result ZEND_FASTCALL add_function(zval *result, zval *op1, zval *op2) /* {{{ */
{
	if (add_function_fast(result, op1, op2) == SUCCESS) {
		return SUCCESS;
	} else {
		return add_function_slow(result, op1, op2);
	}
}
```
and will handle all the edge cases in the same way as an actual ``+`` would do, to which I should actually add a test about adding arrays.

Reply:
Funnily, there really is a `fast_add_function` (not `add_function_fast`). :D

Reply:
> Funnily, there really is a `fast_add_function` (not `add_function_fast`). :D

What? There is a ``add_function_fast()`` it's just a static function in ``zend_operators.c``